homeassistant/components/config/config_entries.py

```diff
@@ -92,6 +93,27 @@ async def delete(self, request, entry_id):
         return self.json(result)
 
 
+class ConfigManagerEntryResourceReloadView(HomeAssistantView):
+    """View to reload a config entry."""
+
+    url = "/api/config/config_entries/entry/{entry_id}/reload"
+    name = "api:config:config_entries:entry:resource:reload"
+
+    async def post(self, request, entry_id):
+        """Reload a config entry."""
+        if not request["hass_user"].is_admin:
+            raise Unauthorized(config_entry_id=entry_id, permission="remove")
+
+        hass = request.app["hass"]
+
+        try:
+            result = await hass.config_entries.async_reload(entry_id)
+        except config_entries.UnknownEntry:
```

Comment:
We should also handle `config_entries.OperationNotAllowed` in case the config entry is in an unrecoverable state (ENTRY_STATE_MIGRATION_ERROR, ENTRY_STATE_FAILED_UNLOAD)

https://developers.home-assistant.io/docs/config_entries_index